torch/testing/_internal/opinfo/core.py

```diff
@@ -1305,21 +1305,18 @@ def get_decorators(self, test_class, test_name, device, dtype, param_kwargs):
         return result
 
     def supported_dtypes(self, device_type):
-        if device_type == "cpu":
-            return self.dtypes
+        device_type = str(device_type).split(":", 1)[0]
         if device_type == "cuda":
             return self.dtypesIfROCM if TEST_WITH_ROCM else self.dtypesIfCUDA
-        else:
-            return self.dtypes
+        return self.dtypes
 
     def supported_backward_dtypes(self, device_type):
         if not self.supports_autograd:
             return set()
 
+        device_type = str(device_type).split(":", 1)[0]
         backward_dtypes = None
-        if device_type == "cpu":
-            backward_dtypes = self.backward_dtypes
```

Comment:
Hint for other reviewers: this branch is covered by the `else` below and thus can be safely removed.

Reply:
Let me add it to the PR description